js/casper.js

```diff
@@ -0,0 +1,54 @@
+var fs = require('fs');
+var links;
+
+function getLinks() {
+    var links = document.querySelectorAll('.nav-link.t-nav-link');
+    return Array.prototype.map.call(links, function (e) {
+        return e.getAttribute('href');
+    });
+}
+
+casper.start('http://192.168.13.37/index.php', function() {
+    fs.write('tests/validation/html_output/index.html', this.getPageContent(), 'w');
+});
+
+casper.then(function () {
+    var links = this.evaluate(getLinks);
+    var current = 0;
+    var end = links.length;
+    // Get the word before .html.twig, the word before the last '/' and the word in urls like '/lexicon'
+    var regex = /((\w*)(\/|.html.twig)$)|(\/\w*$)/gm;
+    var match = [];
+    var newlinks = [];
+
+    // Remove links within the same page
+    for (var i = 0; i < links.length; i++) {
+        if (links[i].indexOf('#')  == -1) {
+            newlinks.push(links[i]);
+        };
+    };
+
+    // Get Page Name from URL
+    for (var i = 0; i < newlinks.length; i++) {
+        if (newlinks[i].match(regex) != null ) { // Remove null items that didn't pass the regex
```

Comment:
This could be simplified further to `if (newlinks[i].match(regex ))` as we'll get an array with the matching regex query if we've got a match, this will also keep the hound at bay.